tls/s2n_async_pkey.c

```diff
@@ -158,14 +158,38 @@ S2N_RESULT s2n_async_pkey_decrypt(struct s2n_connection *conn, struct s2n_blob *
     return S2N_RESULT_OK;
 }
 
+S2N_RESULT s2n_async_cb_execute(struct s2n_connection *conn, struct s2n_async_pkey_op **owned_op)
+{
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE_REF(owned_op);
+    RESULT_ENSURE(conn->handshake.async_state == S2N_ASYNC_NOT_INVOKED, S2N_ERR_ASYNC_MORE_THAN_ONE);
```

Comment:
Do we have a test for this error condition?

Reply:
It's not a new error condition, I just moved it into a helper function. It doesn't look like there's a test.

Reply:
Looks like it's probably not tested because it's difficult to trigger. If the state isn't S2N_ASYNC_NOT_INVOKED, we shouldn't be able to get this far in the code. It's defense in depth.